### PR TITLE
Automatically delete unused dart deps from flutter/DEPS file

### PR DESCRIPTION
**Pull request by aam**
This requires introducing markers to flutter DEPS file as follows:
```
diff --git a/DEPS b/DEPS
index d17dfbc35..ff704a697 100644
--- a/DEPS
+++ b/DEPS
@@ -167,15 +167,17 @@ deps = {
   'src/third_party/icu':
    Var('chromium_git') + '/chromium/deps/icu.git' + '@' + 'c56c671998902fcc4fc9ace88c83daa99f980793',
 
-  'src/third_party/dart':
-   Var('dart_git') + '/sdk.git' + '@' + Var('dart_revision'),
-
   'src/third_party/boringssl':
    Var('github_git') + '/dart-lang/boringssl_gen.git' + '@' + Var('dart_boringssl_gen_rev'),
 
   'src/third_party/boringssl/src':
    'https://boringssl.googlesource.com/boringssl.git' + '@' + Var('dart_boringssl_rev'),
 
+  'src/third_party/dart':
+   Var('dart_git') + '/sdk.git' + '@' + Var('dart_revision'),
+
+  # WARNING: Unused Dart dependencies in the list below till "WARNING:" marker are removed automatically - see create_updated_flutter_deps.py.
+
   'src/third_party/dart/third_party/observatory_pub_packages':
    Var('dart_git') + '/observatory_pub_packages.git' + '@' + Var('dart_observatory_pub_packages_rev'),
 
@@ -356,6 +358,8 @@ deps = {
   'src/third_party/dart/third_party/pkg/yaml':
    Var('dart_git') + '/yaml.git' + '@' + Var('dart_yaml_tag'),
 
+  # WARNING: end of dart dependencies list that is cleaned up automatically - see create_updated_flutter_deps.py.
+
   'src/third_party/colorama/src':
    Var('chromium_git') + '/external/colorama.git' + '@' + '799604a1041e9b3bc5d2789ecbd7e8db2e18e6b8',

```

### CODE REVIEW

**Comment by aam** [comment]
cc @bkonyi

File: tools/dart/create_updated_flutter_deps.py [hunk 1]
```diff
@@ -95,15 +93,38 @@ def Main(argv):
...
+      updatedfile.write('\n')
+      i = i + 1
+      while i < len(lines) and (lines[i].startswith("  # WARNING: end of dart dependencies") == 0):
+        i = i + 1
+      for (k, v) in sorted(old_deps.iteritems()):
+        if (k.startswith('src/third_party/dart/')):
+          for (dart_k, dart_v) in (new_deps.iteritems()):
+            dart_k_suffix = dart_k[len('sdk/') if dart_k.startswith('sdk/') else 0:]
+            if (k.endswith(dart_k_suffix)):
+              updated_value = dart_v
```
Comment (bkonyi) [inline tools/dart/create_updated_flutter_deps.py]: Does this do anything? We overwrite `updated_value` in the next line.

Comment (aam) [reply tools/dart/create_updated_flutter_deps.py]: Ah, right!

**Comment by aam** [comment]
Okay, so then we need to land dart sdk with updated quiver var name, then update flutter/DEPS so it has markers and run  the script once so it just reformats everything and stays unchanged in next runs(unless dependencies do change), then land this CL with the script into buildroot.

**Comment by bkonyi** [comment]
> Okay, so then we need to land dart sdk with updated quiver var name, then update flutter/DEPS so it has markers and run the script once so it just reformats everything and stays unchanged in next runs(unless dependencies do change), then land this CL with the script into buildroot.

I'll let you know when the next SDK roll makes its way in so we can do the necessary updates. Should hopefully be in tonight.

**Comment by bkonyi** [comment]
The latest SDK roll has landed with the updated Quiver var name: https://github.com/flutter/engine/pull/8796 Feel free to make the necessary changes to the DEPS file now.

**Comment by aam** [comment]
> Feel free to make the necessary changes to the DEPS file now.

https://github.com/flutter/engine/pull/8800 with the changes

**Comment by bkonyi** [comment]
We should also roll the buildroot version forward once this change lands.

**Comment by aam** [comment]
> We should also roll the buildroot version forward once this change lands.

Right!

**Pull request merged**